riptide-spring-boot-autoconfigure/src/main/java/org/zalando/riptide/autoconfigure/DefaultRiptideConfigurer.java

```diff
@@ -0,0 +1,99 @@
+package org.zalando.riptide.autoconfigure;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.concurrent.TracedExecutorService;
+import io.opentracing.contrib.concurrent.TracedScheduledExecutorService;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.zalando.logbook.Logbook;
+import org.zalando.riptide.failsafe.CircuitBreakerListener;
+import org.zalando.riptide.failsafe.RetryListener;
+import org.zalando.riptide.logbook.LogbookPlugin;
+import org.zalando.riptide.metrics.MetricsPlugin;
+import org.zalando.riptide.opentracing.OpenTracingPlugin;
+
+import java.util.Optional;
+
+import static org.zalando.riptide.autoconfigure.ValueConstants.LOGBOOK_REF;
+import static org.zalando.riptide.autoconfigure.ValueConstants.METER_REGISTRY_REF;
+import static org.zalando.riptide.autoconfigure.ValueConstants.TRACER_REF;
+
+@AllArgsConstructor
+class DefaultRiptideConfigurer {
+    private final ConfigurableListableBeanFactory beanFactory;
+    private final RiptideProperties properties;
+
+    void register() {
+        properties.getClients().forEach(this::configure);
+    }
+
+    private void configure(final String id, final RiptideProperties.Client client) {
+        if (client.getTracing().getEnabled()) {
+            final BeanDefinition tracerRef = getBeanRef(Tracer.class);
+
+            findBeanDefinition(id, TracedExecutorService.class)
+                    .ifPresent(bd -> replaceConstructorArgumentWithBean(bd, TRACER_REF, tracerRef));
+
+            findBeanDefinition(id, OpenTracingPlugin.class)
+                    .ifPresent(bd -> replaceConstructorArgumentWithBean(bd, TRACER_REF, tracerRef));
+
+            findBeanDefinition(id, TracedScheduledExecutorService.class)
+                    .ifPresent(bd -> replaceConstructorArgumentWithBean(bd, TRACER_REF, tracerRef));
+        }
+
+        if (client.getLogging().getEnabled()) {
+            final BeanDefinition logbookRef = getBeanRef(Logbook.class);
+
+            findBeanDefinition(id, LogbookPlugin.class)
+                    .ifPresent(bd -> replaceConstructorArgumentWithBean(bd, LOGBOOK_REF, logbookRef));
+        }
+
+        if (client.getMetrics().getEnabled()) {
+            final BeanDefinition meterRegistryRef = getBeanRef(MeterRegistry.class);
```

Comment:
This may find nothing and return `null` which means we replace the argument with `null` and then startup with a broken setup. Shouldn't this be throwing if e.g. metrics is enabled but no `MeterRegistry` was found.